client/my-sites/earn/components/add-edit-coupon-modal/test/index.jsx

```diff
@@ -0,0 +1,353 @@
+	const getAmountInput = () => screen.getByLabelText( 'Amount' );
+	const getLimitDurationToggle = () => screen.getByRole( 'checkbox', { name: 'Duration' } );
+	const getLimitDurationSelection = () =>
+		screen.getByRole( 'combobox', { name: 'duration selection' } );
+	const getLimitCouponsToggle = () =>
+		screen.getByRole( 'checkbox', { name: 'Limit coupon to specific emails' } );
+	const getLimitCouponsTextInput = () =>
+		screen.getByRole( 'textbox', { name: 'limit coupon to specific emails text input' } );
+
+	beforeEach( () => {
+		jest.clearAllMocks();
+
+		useTranslate.mockImplementation( mockUseTranslate );
+		translate.mockImplementation( mockTranslate );
+		__.mockImplementation( mockTranslate );
+		getconnectedAccountDefaultCurrencyForSiteId.mockImplementation(
+			mockGetConnectedAccountDefaultCurrencyForSiteId
+		);
+		getconnectedAccountMinimumCurrencyForSiteId.mockImplementation(
+			mockGetConnectedAccountMinimumCurrencyForSiteId
+		);
+
+		modalRoot = document.createElement( 'div' );
+		modalRoot.setAttribute( 'id', 'wpcom' );
+		document.body.appendChild( modalRoot );
+		Modal.setAppElement( modalRoot );
+	} );
+
+	afterEach( () => {
+		unmountComponentAtNode( modalRoot );
+		document.body.removeChild( modalRoot );
+		[ ...document.getElementsByClassName( 'ReactModalPortal' ) ].map( ( el ) => {
+			document.body.removeChild( el );
+		} );
+		modalRoot = null;
+		global.Date = RealDate;
+	} );
+
+	test( 'should render empty form when no coupon is provided', () => {
+		mockDate( '2023-06-23T14:23:44z' );
+		render(
+			<WrappedRecurringPaymentsCouponAddEditModal closeDialog={ closeDialog } coupon={ {} } />
+		);
+		const codeTextbox = getCodeTextbox();
+		const amountOption = getAmountOption();
+		const percentageOption = getPercentageOption();
+		const amountInput = getAmountInput();
+		const limitDurationToggle = getLimitDurationToggle();
+		const limitDurationSelection = getLimitDurationSelection();
+		const limitCouponsToggle = getLimitCouponsToggle();
+		const limitCouponsTextInput = getLimitCouponsTextInput();
+
+		expect( codeTextbox.value ).toEqual( '' );
+		expect( amountOption.selected ).toBe( false );
+		expect( percentageOption.selected ).toBe( true );
+		expect( amountInput ).name = 'discount_percentage';
+		expect( amountInput ).type = 'text';
+		expect( amountInput ).value = '';
+		expect( limitDurationToggle ).not.toBeChecked();
+		expect( limitDurationSelection ).toBeDisabled();
+		expect( limitCouponsToggle ).not.toBeChecked();
+		expect( limitCouponsTextInput ).toBeDisabled();
+	} );
+
+	test( 'should generate a new coupon code when the random button is clicked', () => {
+		mockDate( '2023-06-23T14:23:44z' );
+		render(
+			<WrappedRecurringPaymentsCouponAddEditModal closeDialog={ closeDialog } coupon={ {} } />
+		);
+		const randomButton = getRandomButton();
+		const codeTextbox = getCodeTextbox();
+
+		expect( codeTextbox.value ).toEqual( '' );
+		waitFor( () => randomButton.click() );
```

Comment:
Not blocking as I'm not up to date with Calypso's testing setup, but I'd usually recommend using [`@testing-library/user-event`](https://testing-library.com/docs/user-event/intro/) to simulate user interactions like this one.

If that's not available, is there a reason why we're not using [`fireEvent`](https://testing-library.com/docs/dom-testing-library/api-events/),  instead of generating a synthetic `click` event ?

Reply:
We should use `@testing-library/user-event` indeed 👍

Reply:
Thanks! This is much cleaner!